moirae/extractors/ecm.py

```diff
@@ -210,15 +206,18 @@ def interpolate_r0(self, cycle: pd.DataFrame) -> np.ndarray:
         """
         # calculate soc throughout cycle
         cycle = cycle.copy(deep=False)
-        cycle = self.soc_calc(cycle)
+        if 'cycle_capacity' not in cycle.columns:
+            StateOfCharge().enhance(cycle)

```

Comment:
I have concerns about the methods used to calculate the change in capacity and energy within a given cycle within the Battery Data Toolkit. I will make an issue over there.

Reply:
Absolutely. We do need to polish that algorithm up.

Reply:
Are you cool with using the battery-data-toolkit implementation until we improve it?